lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_const_border_radius/prefer_const_border_radius.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../../../../../utils/node_utils.dart';
+import '../../../lint_utils.dart';
+import '../../../models/internal_resolved_unit_result.dart';
+import '../../../models/issue.dart';
+import '../../../models/replacement.dart';
+import '../../../models/severity.dart';
+import '../../models/flutter_rule.dart';
+import '../../models/rule_documentation.dart';
+import '../../rule_utils.dart';
+
+part 'visitor.dart';
+
+class PreferConstBorderRadiusRule extends FlutterRule {
+  static const ruleId = 'prefer_const_border_radius';
+  static const _issueMessage = 'Prefer use const constructor BorderRadius.all';
+  static const _replaceComment = 'Replace with const BorderRadius constructor';
+
+  PreferConstBorderRadiusRule([Map<String, Object> config = const {}])
+      : super(
+          id: ruleId,
+          documentation: const RuleDocumentation(
+            name: 'Prefer const border radius',
+            brief: 'Warns when used non const border radius',
+          ),
+          severity: readSeverity(config, Severity.performance),
+          excludes: readExcludes(config),
+        );
+
+  @override
+  Iterable<Issue> check(InternalResolvedUnitResult source) {
+    final visitor = _Visitor();
+    source.unit.visitChildren(visitor);
+
+    return visitor.declarations
+        .map((declaration) => createIssue(
+              rule: this,
+              location: nodeLocation(
+                node: declaration,
+                source: source,
+                withCommentOrMetadata: true,
+              ),
+              message: _issueMessage,
+              replacement: _getValueFromAstNode(declaration) != null
+                  ? Replacement(
+                      comment: _replaceComment,
+                      replacement: _getReplacementValue(declaration),
+                    )
+                  : null,
+            ))
+        .toList(growable: false);
+  }
+
+  String _getReplacementValue(AstNode declaration) {
+    final value = _getValueFromAstNode(declaration);
+
+    return 'BorderRadius.all(Radius.circular($value))';
+  }
+}
+
+String? _getValueFromAstNode(AstNode borderRadius) {
+  final paramsList = borderRadius.childEntities;
+
+  return paramsList.isNotEmpty
+      ? _getValueFromString(paramsList.last.toString())
+      : null;
+}
+
+String? _getValueFromString(String value) =>
+    value.length >= 3 && value.startsWith('(') && value.endsWith(')')
+        ? value.substring(1, value.length - 1)
+        : null;
```

Comment:
It's not clear at all what you're trying to achieve by `_getValueFromAstNode` and `_getValueFromString` especially, since you're checking the result for null at line 46. Could you provide more details, what's going on here?

Reply:
I check if there is a parameter in the `BorderRadius.circular(**8**)` constructor. Then I take it out and put it in autofix `BorderRadius.all(Radius.circular(**8**))`. If there is no such parameter, then I do not suggest autofix

Reply:
Okay, let's instead make it a bit different:

```dart
String? _getConstructorArgumentValue(InstanceCreationExpression expression) {
  final arguments = expression.argumentList.arguments;

  return arguments.isNotEmpty
      ? arguments.first.value.toString()
      : null;
}
```

And remove `_getValueFromString` at all.

Reply:
As for creating replacements - create a separate method where you will check for value and return a replacement or null. Right now you call `_getValueFromAstNode` twice for no reason. For instance,

```dart
Replacement? _createReplacement(InstanceCreationExpression expression) {
  final value = _getConstructorArgumentValue(expression);
  
  return value != null
     ? Replacement(
         comment: _replaceComment,
         replacement: 'BorderRadius.all(Radius.circular($value))',
       )
     : null
}
```